lib/ecto/adapters/tds/connection.ex

```diff
@@ -300,10 +300,74 @@ if Code.ensure_loaded?(Tds) do
     end
 
     @impl true
-    def explain_query(_conn, _query, _params, _opts) do
-      raise Tds.Error, "EXPLAIN is not supported by Ecto.Adapters.TDS at the moment"
+    def explain_query(conn, query, params, opts) do
+      params = prepare_params(params)
+
+      case Tds.query_multi(conn, build_explain_query(query), params, opts) do
+        {:ok, [_, %Tds.Result{} = result, _]} ->
```

Comment:
Not sure if the order is deterministic, but looks like it's (still need to confirm on TDS code).